### PR TITLE
Datadog Agent v6 Changes

### PR DESCRIPTION
**Pull request by ruthnaebeck**
### What does this PR do?
Format Agent FAQ index like other FAQ index pages

Migrate Agent change docs from the `datadog-agent` repo:
* https://github.com/DataDog/datadog-agent/blob/master/docs/agent/changes.md
* https://github.com/DataDog/datadog-agent/blob/master/docs/agent/config.md
* https://github.com/DataDog/datadog-agent/blob/master/docs/agent/missing_features.md

### Motivation
- Single source of documentation

### Preview link
https://docs-staging.datadoghq.com/ruth/agent-changes/agent/faq/agent-v6-changes/

### Additional Notes
- Add placeholders to `datadog-agent` repo after this is merged.

### CODE REVIEW

File: content/en/agent/faq/agent-v6-changes.md [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+#### Configuration options
+
+The following Agent configuration options were changed or removed in Agent v6. Configuration options removed were either superseded by other options, or related to features that work differently from previous versions.
+
+##### Changed
+
+| Previous Name               | Updated Name                 | Notes                                                                                             |
+|-----------------------------|------------------------------|---------------------------------------------------------------------------------------------------|
+| `proxy_host`                | `proxy`                      | Proxy settings are now expressed as a list of URIs. See the [proxy][3] documentation for details. |
+| `collect_instance_metadata` | `enable_metadata_collection` | Enables metadata collection                                                                       |
```
Comment (l0k0ms) [inline content/en/agent/faq/agent-v6-changes.md]:
```suggestion
| `collect_instance_metadata` | `enable_metadata_collection` | Enables metadata collection.                                                                       |
```

File: content/en/agent/faq/agent-v6-changes.md [hunk 2]
```diff
@@ -0,0 +1,564 @@
...
+| `collector_log_file`        | `log_file`                   |                                                                                                   |
+| `syslog_host`               | `syslog_uri`                 | The Syslog configuration is now expressed as a URI.                                               |
+|                             | `syslog_pem`                 | Syslog configuration client certificate for TLS client validation.                                |
+|                             | `syslog_key`                 | Syslog configuration client private key for TLS client validation.                                |
```
Comment (l0k0ms) [inline content/en/agent/faq/agent-v6-changes.md]: There is no previous name for those two parameters ?

File: content/en/agent/faq/agent-v6-changes.md [hunk 3]
```diff
@@ -0,0 +1,564 @@
...
+| `use_curl_http_client`       | Obsolete                                                                                                              |
+| `exclude_process_args`       | Deprecated feature                                                                                                    |
+| `check_timings`              | Superseded by internal stats                                                                                          |
+| `non_local_traffic`          | Superseded by `dogstatsd_non_local_traffic` for Dogstatsd and `apm_config.apm_non_local_traffic` for the Trace Agent. |
+| `dogstatsd_target`           |                                                                                                                       |
+| `dogstreams`                 | Deprecated feature, use the [Logs Agent][5] instead.                                                                  |
+| `custom_emitters`            |                                                                                                                       |
```
Comment (l0k0ms) [inline content/en/agent/faq/agent-v6-changes.md]: What happened to this parameter ?

File: content/en/agent/faq/agent-v6-changes.md [hunk 4]
```diff
@@ -0,0 +1,564 @@
...
+| `forwarder_log_file`         | Superseded by `log_file`                                                                                              |
+| `dogstatsd_log_file`         | Superseded by `log_file`                                                                                              |
+| `jmxfetch_log_file`          | Superseded by `log_file`                                                                                              |
+| `syslog_port`                | Superseded by `syslog_uri`                                                                                            |
+| `check_freq`                 |                                                                                                                       |
+| `collect_orchestrator_tags`  | Implemented in metadata collectors                                                                                    |
+| `utf8_decoding`              |                                                                                                                       |
```
Comment (l0k0ms) [inline content/en/agent/faq/agent-v6-changes.md]: Same question, should we just say they are now `Deprecated feature` ?

File: content/en/agent/faq/agent-v6-changes.md [hunk 5]
```diff
@@ -0,0 +1,564 @@
...
+Agent v6 loads any valid YAML file in: `<AGENT_DIRECTORY>/conf.d/<CHECK_NAME>.d/`. This enables complex configurations to be broken down into multiple files.
+
+For example, configuration files for the `http_check` could be:
+```text
+/etc/datadog-agent/conf.d/http_check.d/
+├── backend.yaml
+└── frontend.yaml
+```
+
+The Agent won't load configuration files from any sub-directories within the `<CHECK_NAME>.d` folder. For example, this configuration is **NOT** loaded:
```
Comment (l0k0ms) [inline content/en/agent/faq/agent-v6-changes.md]:
```suggestion
The Agent doesn't load configuration files from any sub-directories within the `<CHECK_NAME>.d` folder. For example, this configuration is **NOT** loaded:
```

File: content/en/agent/faq/agent-v6-changes.md [hunk 6]
```diff
@@ -0,0 +1,564 @@
...
+```text
+/etc/datadog-agent/conf.d/http_check.d/prod.d/
+├── backend.yaml
+```
+
+Autodiscovery template files (`auto_conf.yaml`) are stored in the configuration folder as well. This is an example of the `redisdb` check configuration folder:
```
Comment (l0k0ms) [inline content/en/agent/faq/agent-v6-changes.md]: We should add a link to the Autodiscovery doc section here.

File: content/en/agent/faq/agent-v6-changes.md [hunk 7]
```diff
@@ -0,0 +1,564 @@
...
+```text
+/etc/datadog-agent/conf.d/redisdb.d/
+├── auto_conf.yaml
+└── conf.yaml.example
+```
+
+The YAML files within a `<CHECK_NAME>.d` folder can have any name, as long as they have a `.yaml` or `.yml` extension. The standard name is `conf.yaml`.
+
+To keep backwards compatibility, the Agent still picks up configuration files in the form: `<AGENT_DIRECTORY>/conf.d/<CHECK_NAME>.yaml` but migrating to the updated layout is strongly recommended.
```
Comment (l0k0ms) [inline content/en/agent/faq/agent-v6-changes.md]:
```suggestion
To keep backwards compatibility, the Agent picks up configuration files in the form: `<AGENT_DIRECTORY>/conf.d/<CHECK_NAME>.yaml` but migrating to the updated layout is strongly recommended.
```

File: content/en/agent/faq/agent-v6-changes.md [hunk 8]
```diff
@@ -0,0 +1,564 @@
...
+
+#### Configuration options
+
+Agent v6 supports the following advanced options in a check's `instance` section:
```
Comment (l0k0ms) [inline content/en/agent/faq/agent-v6-changes.md]:
```suggestion
Agent v6 supports the following options in a check's `instance` section:
```

File: content/en/agent/faq/agent-v6-changes.md (hunk 2 above)
Comment (ruthnaebeck) [reply content/en/agent/faq/agent-v6-changes.md]: Not that was available in the current docs

File: content/en/agent/faq/agent-v6-changes.md (hunk 3 above)
Comment (ruthnaebeck) [reply content/en/agent/faq/agent-v6-changes.md]: It's in the removed list but there were no notes about it on current doc

File: content/en/agent/faq/agent-v6-changes.md (hunk 4 above)
Comment (ruthnaebeck) [reply content/en/agent/faq/agent-v6-changes.md]: It's in the removed list but there were no notes about it on current doc

File: content/en/agent/faq/agent-v6-changes.md [hunk 9]
```diff
@@ -0,0 +1,564 @@
...
+If the `service` wrapper command is not available on your system, use:
+
+* On `upstart`-based systems: `sudo start/stop/restart/status datadog-agent`
+* On `systemd`-based systems: `sudo systemctl start/stop/restart/status datadog-agent`
+
+If you're unsure which init system your distribution uses by default, refer to the table below:
+
+| distribution \ init system      | upstart                   | systemd                   | sysvinit                                  | Notes                         |
+|---------------------------------|---------------------------|---------------------------|-------------------------------------------|-------------------------------|
+| Amazon Linux (<= 2017.09)       | <i class="icon-tick"></i> |                           |                                           |                               |
```
[the remark this answers is not in the document]
Comment (ruthnaebeck) [reply content/en/agent/faq/agent-v6-changes.md]: This won't work properly inside the tab which is why I used `<i class="icon-tick"></i>`.

**Pull request merged**